core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/SPARQLQueries.java

```diff
@@ -0,0 +1,131 @@
+/******************************************************************************* 
+ * Copyright (c) 2021 Eclipse RDF4J contributors. 
+ * All rights reserved. This program and the accompanying materials 
+ * are made available under the terms of the Eclipse Distribution License v1.0 
+ * which accompanies this distribution, and is available at 
+ * http://www.eclipse.org/org/documents/edl-v10.php. 
+ *******************************************************************************/
+package org.eclipse.rdf4j.query.parser.sparql;
+
+import org.eclipse.rdf4j.common.text.StringUtil;
+import org.eclipse.rdf4j.model.Namespace;
+
+/**
+ * Utility functions for working with SPARQL query strings.
+ * 
+ * @author Jeen Broekstra
+ */
+public class SPARQLQueries {
+
+	/**
+	 * Creates a string representing of the supplied {@link Namespace namespaces} as SPARQL prefix declarations. This
+	 * can be used when composing a SPARQL query string in code, for example:
+	 * 
+	 * <pre>
+	 * <code>
+	 * String query = SPARQLQueries.getPrefixClauses(connection.getNamespaces()) + "SELECT * WHERE { ?s ex:myprop ?o }";
+	 * </code>
+	 * </pre>
+	 * 
+	 * @param namespaces one or more {@link Namespace} objects.
+	 * @return one or more SPARQL prefix declarations (each separated by a newline), as a String.
+	 * 
+	 * @since 3.6.0
+	 */
+	public static String getPrefixClauses(Iterable<Namespace> namespaces) {
+		final StringBuilder sb = new StringBuilder();
+		for (Namespace namespace : namespaces) {
+			sb.append(String.format("PREFIX %s: <%s>\n", namespace.getPrefix(), namespace.getName()));
+		}
+		return sb.toString();
+	}
+
+	/**
+	 * Escape the supplied string with backslashes for any special characters, so it can be used as a string literal
+	 * value in a SPARQL query.
+	 * 
+	 * @see https://www.w3.org/TR/sparql11-query/#grammarEscapes
+	 * 
+	 * @since 3.6.0
+	 */
+	public static String escape(String s) {
```

Comment:
I'm a little bit uncertain about keeping this method as-is: while it handles string literal escaping as describe in section 19.4 (https://www.w3.org/TR/sparql11-query/#grammarEscapes), it does not handle the more general unicode code point escaping (see https://www.w3.org/TR/sparql11-query/#codepointEscape) .  We have functionality for that elsewhere but we may want to make it more obvous which method(s) to apply where. It goes a bit beyond the scope of this PR to fully fix that though.

Reply:
Up to you.

Reply:
Well, if it's documented in the code that it may not work in all cases... then I'd say "don't let perfect be the enemy of the good"

Reply:
Thanks, we'll keep as-is then.